acceptance/api/v1/services_test.go

```diff
@@ -260,4 +264,125 @@ var _ = Describe("Services API Application Endpoints", func() {
+			Expect(service).ToNot(HaveKey("username"))
+		})
+
+		It("returns a 404 when the namespace does not exist", func() {
+			response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/idontexist/services/%s",
+				serverURL, api.Root, svc1), strings.NewReader(changeRequest))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+		})
+
+		It("returns a 404 when the service does not exist", func() {
+			response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/bogus",
+				serverURL, api.Root, namespace), strings.NewReader(changeRequest))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response).ToNot(BeNil())
+
+			defer response.Body.Close()
+			bodyBytes, err := ioutil.ReadAll(response.Body)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(response.StatusCode).To(Equal(http.StatusNotFound), string(bodyBytes))
+		})
+
+		When("service is bound to an app", func() {
+			var app1 string
+			BeforeEach(func() {
+				app1 = catalog.NewAppName()
+
+				env.MakeContainerImageApp(app1, 1, containerImageURL)
+				env.BindAppService(app1, svc1, namespace)
+			})
+
+			AfterEach(func() {
+				env.DeleteApp(app1)
+			})
+
+			Describe("workload restarts", func() {
+				It("only restarts the app if the service has changed", func() {
+					getPodName := func(namespace, app string) (string, error) {
+						return helpers.Kubectl("get", "pods", "-n", namespace, "-l", fmt.Sprintf("app.kubernetes.io/name=%s", app), "-o", "jsonpath='{.items[*].metadata.name}'")
+					}
+
+					oldPodName, err := getPodName(namespace, app1)
+					Expect(err).ToNot(HaveOccurred())
+
+					response, err := env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/%s",
+						serverURL, api.Root, namespace, svc1), strings.NewReader(changeRequest))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).ToNot(BeNil())
+					Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+					Eventually(func() string {
+						newPodName, err := getPodName(namespace, app1)
+						Expect(err).ToNot(HaveOccurred())
+						return newPodName
+					}, "2m").ShouldNot(Equal(oldPodName))
+
+					// Now try with no changes
+					oldPodName, err = getPodName(namespace, app1)
+					Expect(err).ToNot(HaveOccurred())
+
+					response, err = env.Curl("PUT", fmt.Sprintf("%s%s/namespaces/%s/services/%s",
+						serverURL, api.Root, namespace, svc1), strings.NewReader(changeRequest))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(response).ToNot(BeNil())
+					Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+					Consistently(func() string {
+						newPodName, err := getPodName(namespace, app1)
+						Expect(err).ToNot(HaveOccurred())
+						return newPodName
+					}, "30s").Should(Equal(oldPodName))
```

Comment:
https://onsi.github.io/gomega/#consistently

`Consistently` will run for the whole period defined (here: 30s). It's supposed to check that it doesn't change during that time. I think 30s is too much in this case. Also [the test fails](https://github.com/epinio/epinio/runs/4545253910?check_suite_focus=true) which seems as if a restart actually happened.